### DISCUSSION TITLE
Copy related object property on creation

### DISCUSSION THREAD

**Discussion by tasops** [prisma/prisma#4860] [Q&A] [answered]
Shalom! I don't know how to word that problem exactly, so let me show you what i want to accomplish.

Here are example models:
```prisma
model Product {
  id         Int              @id @default(autoincrement())
  name       String
  price      Float
  orderItems OrderItem[]
}

model OrderItem {
  id             Int     @id @default(autoincrement())
  product        Product @relation(fields: [productId], references: [id])
  priceAtBuyTime Float
  order          Order   @relation(fields: [orderId], references: [id])
  productId      Int
  orderId        Int
}

model Order {
  id    Int             @id @default(autoincrement())
  items OrderItem[]
}
```

And here is an example of what i want to do:
```ts
const exampleRequestBody = [1, 2, 3]

const order = await prisma.order.create({
  items: {
    create: exampleRequestBody.map(id => ({
      product: {
        connect: { id }
      },
      priceAtBuyTime: // Somehow get the price of the product connected above
    }))
  }
})
```

In short, i want to copy that price to that object because the price may change, and i want to take a 'snapshot' so then the user can see his order in the future - with the price that he paid.

**Comment by ryands17** [comment, marked as answer]
Hey @Telluu 👋 

I didn't quite get your question. Do you want to add the value that you just created in the `priceAtBuyTime` field or do you want to add the value to the same field of the `Order` model?

**Reply by tasops** [reply]
So when I create a new `Order`, that order has many `OrderItem`, that single `OrderItem` then connects to some `Product` model that i created before. 

When i connect that `Product`, i need to get the price of that `Product` and paste it to `OrderItem` field `priceAtBuyTime`.

Here is something that i found and describes my problem as a whole: 
https://stackoverflow.com/questions/6233846/updating-product-prices-changes-pricing-on-all-orders

There are two solutions:
1. One that i want to solve right now, by just "hard coding" it as a string field.
2. Or more elegant one, that relies on relations. A price history table. 

Tried to make it as simple as possible. If you have any questions, ask. I may just make that price history table and do it the right way if there is no simple way of doing the first solution.

**Reply by ryands17** [reply]
I think in this case, the best way to go would be the price history table which would be better as you would be able to fetch the correct value every time instead of hardcoding :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
